### DISCUSSION TITLE
Flow-level settings for users

### DISCUSSION THREAD

**Discussion by multimeric** [PrefectHQ/prefect#5636] [Q&A]
So the Prefect config file provides a very neat mechanism to allow flow authors to define their own custom config (under a custom key), and then the end user can configure both the Prefect engine, and the flow's settings in the same file. I'm firstly wondering if this is intended behaviour, since it's not actually explained anywhere in the docs that the `config.toml` will accept arbitrary keys.

If it is intended, I'm wondering if we could better support this workflow, by letting the flow authors define default values, and also specifying which keys are required. The same kinds of things you can do for a CLI using argparse.

If not, is there a recommended way to allow users to pass flow-level settings rather than Prefect-level settings? Does this method support the validations I mention above?

**Comment by zanieb** [comment]
This is an interesting observation! I don't know of anyone doing this in practice, but it's a cool idea. What kind of default values are you setting with this? Why do flow parameters not suffice?

**Reply by multimeric** [reply]
Perhaps I'm misunderstanding, but it seems that the only way for users to provide parameter values are either at runtime, or by editing the flow. Neither suits my use case, because I have so many parameters and I want the user to be able to tweak them in a config file instead of bugging them for 20+ different values. 

I'm wondering if a neater solution for Prefect would be to add a new subtable in the config file called `[parameters]` that allows users another mechanism for providing parameters. Then the normal default/required mechanisms for parameters can be used for that instead.

**Reply by zanieb** [reply]
Hm, I'm not sure introducing another source for parameters would make sense across all of our use-cases. Have you considered having the first task in your flow just read a file to pull the values you need?

**Reply by multimeric** [reply]
Fair enough. Well for now I'm happy to abuse the config file for this purpose. It seems to work quite well in conjunction with pydantic.
